### PR TITLE
[codestyle] works better with J2EE

### PR DESCRIPTION
**Pull request by akshitdewan**
Might be surprised by how classLoader() works with J2EE, improves some test cases

### CODE REVIEW

File: xwiki-rendering-test/src/main/java/org/xwiki/rendering/test/cts/TestDataParser.java [hunk 1]
```diff
@@ -75,7 +75,7 @@
     public List<TestData> parseTestData(String syntaxId, String ctsRootPackageName, String packageFilter,
         String pattern) throws Exception
     {
-        ClassLoader classLoader = getClass().getClassLoader();
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
```
Comment (vmassol) [inline xwiki-rendering-test/src/main/java/org/xwiki/rendering/test/cts/TestDataParser.java]: hmm I don't think this is right. The idea was to use the classloader used by the TestDataParser class to load the other classes. This is used when tests are executing. I don't think the current context CL is even set at that time. Did you try the change in real?

Comment (vmassol) [reply xwiki-rendering-test/src/main/java/org/xwiki/rendering/test/cts/TestDataParser.java]: (this is not JavaEE code :))

Comment (tmortagne) [reply xwiki-rendering-test/src/main/java/org/xwiki/rendering/test/cts/TestDataParser.java]: This time this change is a mistake. We explicitly want to make sure to use the classloader of the test class to make sure to not end up with unexpected class.

In other words you found a false positive.

Comment (akshitdewan) [reply xwiki-rendering-test/src/main/java/org/xwiki/rendering/test/cts/TestDataParser.java]: alright I'll put it on the list

**Pull request closed without merging**